src/main/java/com/google/sps/model/PlacesClientImpl.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.maps.GeoApiContext;
+import com.google.maps.NearbySearchRequest;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceType;
+import com.google.maps.model.PlacesSearchResponse;
+import com.google.maps.model.RankBy;
+import com.google.sps.exceptions.PlacesException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** Handles GET requests to the Google Places API */
+public class PlacesClientImpl implements PlacesClient {
+  private NearbySearchRequest placesService;
+
+  private PlacesClientImpl(NearbySearchRequest service) {
+    placesService = service;
+  }
+
+  /** Factory to create a PlacesClientImpl instance with given API key */
+  public static class Factory implements PlacesClientFactory {
+    @Override
+    public PlacesClient getPlacesClient(String apiKey) {
+      GeoApiContext context = new GeoApiContext.Builder().apiKey(apiKey).build();
+      NearbySearchRequest service = new NearbySearchRequest(context);
+      return new PlacesClientImpl(service);
+    }
+  }
+
+  /**
+   * Gets all formatted addresses from given response. Scope of method is public for testing
+   * purposes.
+   *
+   * @param result The PlacesSeachResponse object to get formatted addresses from
+   */
+  public static List<String> getFormattedAddresses(PlacesSearchResponse response) {
```

Comment:
I would move this to the PlacesClient.java interface (which can contain static methods), since this pertains to the overall function of any PlacesClient, not just the impl

Reply:
Also javadoc should include some information about what the format of a formatted address is

Reply:
Made the changes! Let me know if it's better now.

Reply:
`RESOLVED`